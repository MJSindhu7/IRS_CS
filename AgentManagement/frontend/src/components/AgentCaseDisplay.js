import React, { Component,useState } from "react";
import axios from "axios";
import {
    Button,
    Row,
    Col,
    Container,
    Tabs,
    Tab,
    Nav
  } from "react-bootstrap";
import { Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import '../css/agent.css';
import AgentCaseHistory from './AgentCaseHistory';
import Messages from './Messages';
import {} from "./utils.js";
import swal from 'sweetalert'
import {getEmailId, getAgentId, getOrganisationId, removeOrgId} from './utils';
import AddAgentMessage from './AddAgentMessage'

const config = require("../config/settings.js");

class AgentCaseDisplay extends Component{
    constructor(props){
        super(props);
       
        this.state={
            caseDetails: props.caseDetails,
            history:"",
            caseId:"",
            isLoaded: false,
            activeKey:'caseDetails'
        }
    }
  
    handleClose = () => this.setState({show:false, activeKey: "caseDetails"});
    handleShow = () => this.setState({show:true, activeKey: "caseDetails"});
    componentDidMount(){
        this.getHistory(this.props.caseDetails)
     
    }
    componentDidUpdate(prevProps){
      if(prevProps.caseId != this.props.caseId){
        this.setState({
          caseDetails: this.props.caseDetails,
          caseId: this.props.caseId,
          isLoaded: false,
          activeKey: "caseDetails"
        })
        this.getHistory(this.props.caseDetails)
      }
    }
    changeActiveKey = (key) =>{
        console.log("active key:", key)
        this.setState({
          activeKey: key
        })
      }

    
    getHistory = async ()=>{
      let {caseDetails} = this.props
        this.setState({
            isLoaded: false
        })
      if(caseDetails && caseDetails.CaseID && caseDetails.UserID){
        await axios(config.rooturl + "/history/" + caseDetails.UserID +"/"+caseDetails.CaseID  , {
            method: "get",
            config: { headers: { "Content-Type": "application/json" } }
          })
            .then((res) => {
              this.setState({ history: res.data, isLoaded:true });
              console.log("history: ", this.state.history);
            })
            .catch((error) => {
                console.log(error.response.data)
                this.setState({
                    isLoaded: true
                })
            });
      }
    }

    render(){
       let {caseDetails, isLoaded} = this.state;
       const closeBtn = (
        <button className="close" onClick={() => this.props.showModal()}>
          &times;
         </button>
      );
      if(caseDetails){
        return (
              <Modal  
              isOpen={this.props.modal}
              toggle={() => this.props.showModal()}
              className="modal-lg"
              scrollable>
                <ModalHeader
                      size="lg"
                      toggle={() => this.props.showModal()}
                      close={closeBtn}
                    >
                    <div className="flex">
                    <span>Case ID: {caseDetails.CaseID}</span>
                    </div>
                </ModalHeader>
                <ModalBody>
                    <Tabs activeKey={this.state.activeKey} id="casetab" onSelect={(k) => this.changeActiveKey(k)}>
                        <Tab eventKey="caseDetails" title="Case Details">
                           <CaseDetails caseDetails = {caseDetails} changeActiveKey = {this.changeActiveKey} handleClose={this.handleClose} getHistory={this.getHistory}/>
                        </Tab>
                        <Tab eventKey="history" title="Case History">
                            {/*<CaseHistory caseId={caseDetails.CaseID} userId={caseDetails.UserID}/>*/}
                            {isLoaded && <AgentCaseHistory caseId = {this.state.caseId} history={this.state.history}/>}
                        </Tab>
                    </Tabs>
                  
                </ModalBody>
                <ModalFooter>
                </ModalFooter>
              </Modal>
          )
        } else {
          return <div></div>
        }
    }
    
  }
  
export default AgentCaseDisplay;

class CaseDetails extends Component{
    constructor(props){
        super(props);
        this.state = {
            caseDetails: props.caseDetails,
            newStatus: ''
        }
    }

    isNotResolvedCase = () =>{
         let {caseDetails} = this.state;
         console.log("isNotResolvedCase:"+ !(caseDetails.Status && caseDetails.Status.toLowerCase() == 'resolved'))
         return !(caseDetails.Status && caseDetails.Status.toLowerCase() == 'resolved')
    }

    handleChange = (e) => {
        let status = e.target.value;
        this.setState({
          newStatus: status
        });
      }
  
      updateStatus = () =>{
        console.log("In updateStatus in case display")
        console.log(this.state.newStatus);
        let {caseDetails} = this.state
        if(this.state.newStatus !== "" && caseDetails.Status !== this.state.newStatus){
          console.log("in if block")
          console.log("In status")
          console.log(this.state.newStatus)
          console.log("in old status")
          console.log(caseDetails.Status);
          let data = {
            caseID: caseDetails.CaseID,
            status: this.state.newStatus,
            userID: caseDetails.UserID
          }
          axios('http://' + config.hostname + ':' + config.backendPort + '/updateStatus',{
            method: 'post',
            data: data
          }).then((response) => {
          if (response.status == 500) {
            throw new Error("Bad response from server");
          }else{
            console.log("in update status..")
            this.props.getHistory();
            swal("Status updated successfully").then((val)=> this.props.changeActiveKey("history"))
            console.log(response);
          }
        }).catch(function (err) {
            console.log(err)
          });
          caseDetails.Status = this.state.newStatus; 
      }
    }

    render(){
        let {caseDetails} = this.state;
        return(
            <>
             <Container className="caseDetailsTab">
                <Row>
                    <Col><b>Description:</b></Col>
                </Row>
                <Row>
                    <Col>{caseDetails.Information}</Col>
                </Row>
                <br></br>
                <br></br>
                <Row>
                    <Col><b>Status:</b></Col>
                    <Col><b>Category:</b></Col> 
                </Row>
                <Row>
                    {this.isNotResolvedCase() && <Col>
                        <select id="Status" defaultValue={caseDetails.Status} onChange={this.handleChange}>
                        <option value="Assigned">Assigned</option>
                        <option value="InProgress">In Progress</option>
                        <option value="Resolved">Resolved</option>
                        </select>
                    </Col>}

                    {!this.isNotResolvedCase() && <Col>
                       {caseDetails.Status}
                    </Col>}

                    <Col>{removeOrgId(caseDetails.Category)}</Col>
                </Row>
            </Container>
        <br/>
        {this.isNotResolvedCase() && <AddAgentMessage caseId = {caseDetails.CaseID} changeActiveKey = {this.props.changeActiveKey} handleClose={this.props.handleClose} getHistory = {this.props.getHistory} updateStatus = {this.updateStatus}/>}
         </>
        )
    }
}