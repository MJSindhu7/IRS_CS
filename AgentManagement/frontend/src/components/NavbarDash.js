import React, { Component } from 'react';
import logo from './../css/IRS.png'

class NavbarDash extends Component {
    
  handleLogout = () => {
    window.localStorage.clear();
  };

  render() {
    return (
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark mb-4 fixed-top">
        
          <img  src={logo}
           width="33"
           height="30"
           className="d-inline-block align-top"/>
           
          <a className="navbar-brand" href="/AgentDashboard">
            Dashboard
          </a>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon"></span>
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <p className="nav-link" >
                  Hi, {localStorage.getItem("userName")}
                </p>
              </li>
              <li className="nav-item">
                <a className="nav-link" href="/" onClick={this.handleLogout}>
                  Logout
                </a>
              </li>
            </ul>
          </div>
        
      </nav>
    );
  }
}

export default NavbarDash;