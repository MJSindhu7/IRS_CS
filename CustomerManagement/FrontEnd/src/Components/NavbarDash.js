import React, { Component } from 'react';
import {getName} from './utils.js'

class NavbarDash extends Component {
    
  handleLogout = () => {
    window.localStorage.clear();
  };

  render() {
    return (
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark mb-4">
        <div className="container">
          <a className="navbar-brand" href="/AgentDashboard">
            IRS Dashboard
          </a>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon"></span>
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <p className="nav-link" >
                   {getName()}
                </p>
              </li>
              <li className="nav-item">
                <a className="nav-link" href="/OrgOwnerLogin" onClick={this.handleLogout}>
                  Logout
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}

export default NavbarDash;