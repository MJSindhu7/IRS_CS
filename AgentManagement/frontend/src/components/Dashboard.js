import React, { Component } from 'react';
import axios from "axios";
import NavbarDash from "./NavbarDash";
import Sidebar from './Sidebar'
import config from '../config/settings'
import Table from 'react-bootstrap/Table'
import Pagination from 'react-bootstrap/Pagination'
import {getAgentId, getOrganisationId} from './utils'
import AgentCaseDisplay from './AgentCaseDisplay'

class Dashboard extends Component {
  constructor() {
    super();
    this.state = {
      allCases: [],
      matchedCases: [],
      viewcase: null,
      modal: false,
      mstatus: "open",
      flag: false,
      selectOptions: [],
      status: "",
      paginationActive: 1,

    };

  }
  onClickPaginationItem = (e, number) => {
    e.preventDefault();
    // console.log('clicked an item');
    this.setState({
      paginationActive: number,
    })
  }
  
  handleChange = (e) => {
    let status = e.target.value;
    let allCases = this.state.allCases;
    let matchedCases = [];
    if (status == "") {
      matchedCases = allCases;
      status = "";
    } else {
      for (var i = 0; i < allCases.length; i++) {
        if (allCases[i].Status == status) {
          matchedCases.push(allCases[i]);
        }
      }
    }

    this.setState({
      status: status,
      matchedCases: matchedCases,
    });
  }

  showModal = () => {
    console.log("hello");
    this.setState({
      modal: !this.state.modal,
    });
  };

  showModal1 = (viewcase) => {
    console.log("hello1");
    this.setState({
      modal: !this.state.modal,
      viewcase: viewcase,
    });
  };

  handleStatusChange = (modalstatus) => {
    this.setState({
      mstatus: modalstatus
    })
  }

  async componentDidMount() {
    console.log("in get all cases frontend")

   
    axios.defaults.withCredentials = true;
     let agentID = getAgentId();
  
    let organisationID = getOrganisationId()
    let count = 0
    axios.defaults.withCredentials = true;
    if (agentID) {
      count = count + 1
      let data = {
        agentID: agentID,
        organisationID: organisationID
      };
      console.log("data is..")
      console.log(data);

      axios({
        method: 'get',
        url: 'http://' + config.hostname + ':' + config.backendPort + '/getCases',
        params: data,
      })
        .then((response) => {
          console.log("response from getCases")
          console.log(response)

          if (response.status === 200) {
            let receivedCases = response.data.cases;
            console.log("received cases")
            console.log(response.data.cases)
            if (receivedCases) {

              this.setState({
                allCases: receivedCases,
                flag: true,
              });


              const data = this.state.allCases

              const options = data.map(d => ({
                "value": d.Status,
                "label": d.Status
              }))
              this.setState({ selectOptions: options })
              console.log("options")
              console.log(this.state.selectOptions)

            } else {
              this.setState({ allCases: [] });
            }
          }
          else {
            this.setState({ allCases: [] });
          }

        })
        .catch(error => {
          console.log(error);
          this.setState({ allCases: [] })
        })


    }
    else if (count < 1) {
      window.location.reload();
    }
  }


  render() {
    if(!this.state.allCases || this.state.allCases.length == 0){
      return(
        <>
        <NavbarDash />
        <div className="row">
          <div className="col-2">
            <Sidebar />
          </div>
          </div>
          <div className="container col-9" style={{marginTop:'6rem'}}>
          <h4 style={{ margin: "3em" }}>No new cases to display!</h4>
          </div>
      </>
      )
    }
    console.log("status");
    console.log(this.state.status);
    var casedetails;
    let searchResults;
    let status = this.state.status; 
    if (status == "") {
      searchResults = this.state.allCases;
    } else {
      searchResults = this.state.matchedCases;
    }

    let index = 0;
    let maxResultsPerPage = 6;
    let active = this.state.paginationActive;     // Initially active key
    let startIndex = 0 + (active - 1) * maxResultsPerPage;
    let endIndex = startIndex + maxResultsPerPage;


    // Pagination
    let items = [];     // 
    for (let number = 1; number <= Math.ceil(searchResults.length / maxResultsPerPage); number++) {
      let isActive = (number === active);
      items.push(
        <Pagination.Item key={number} active={isActive} onClick={(e) => {
          this.onClickPaginationItem(e, number)
        }}>
          {number}
        </Pagination.Item>,
      );
    }

    const paginationBasic = (
      <div>
        <Pagination>{items}</Pagination>
      </div>
    );

    searchResults = searchResults.slice(startIndex, endIndex);

    if (this.state.flag) {
      casedetails = searchResults.map((ticket) => {
          return (
            <tr onClick={() => this.showModal1(ticket)}>
                  <td>{ticket.CaseID}</td>
                  <td>{ticket.CreatedOn}</td>
                  <td>{ticket.Status}</td>
                  <td>{ticket.Information} </td>
                </tr>
          );
      });
    }
    return (
      <div style ={ { marginTop:'6rem'}} >
        <NavbarDash />
        <div className="row">
          <div className="col-2">
            <Sidebar />
          </div>
          <div className="container col-9">

            <div className="filter-item">
              {this.state.allCases.map(function (item) {
                return (
                  <div>{item.name}</div>
                );
              })}
            </div>
            <div>
              Status: < nbsp />
              <select id="Status" value={this.state.status} onChange={this.handleChange}>
                <option value="">All</option>
                <option value="Assigned">Assigned</option>
                <option value="InProgress">In Progress</option>
                <option value="Resolved">Resolved</option>
              </select>
            </div>
              <br></br>
              <div style={{ minHeight: '600px' }}>  
            <Table striped bordered hover cursorPointer>
              <tr>
                <th style={{ width: '10rem' }}>ID</th>
                <th style={{ width: '10rem' }}>Date</th>
                <th style={{ width: '15rem' }}>Status</th>
                <th style={{ width: '50rem' }}>Details</th>
              </tr>
              <tbody>
              {casedetails}
              </tbody>
            </Table>
            </div>
            {this.state.allCases.length > 0 ? (
              <div className="col-10">
              
                {this.state.viewcase != null ? (
                  <div>
                  <AgentCaseDisplay caseId = {this.state.viewcase.CaseID}
                  showModal = {this.showModal} modal = {this.state.modal} caseDetails={this.state.viewcase}/>
                 </div> ) : null}
                 
               <div  style = { {  height: '15%',  display: "flex",justifyContent: "center",alignItems: "center"}}>
                 {paginationBasic}
                </div>
                
                 
                
                </div>
            ) : (
                <div>
                  <h4 style={{ margin: "3em" }}>No new cases to display!</h4>
                </div>
              )}
              
              
          </div>
        
              
        </div>
           
      </div>
    );
  }
}

export default Dashboard;
