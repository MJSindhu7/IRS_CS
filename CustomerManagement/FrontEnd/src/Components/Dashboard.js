import React, { Component } from 'react';
import axios from "axios";
import NavbarDash from "./NavbarDash";
import { Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import Sidebar from './Sidebar'
import config from '../config/settings'
import Card from 'react-bootstrap/Card'
import Dropdown from 'react-bootstrap/Dropdown'
import Table from 'react-bootstrap/Table'

class Dashboard extends Component {
  constructor() {
    super();
    this.state = {
      allCases: {},
      viewcase: null,
      modal: false,
      mstatus: "open",
      flag: false
    };

  }

  showModal = () => {
    console.log("hello");
    this.setState({
      modal: !this.state.modal,
    });
  };
  showModal1 = (viewcase) => {
    console.log("hello1");
    this.setState({
      modal: !this.state.modal,
      viewcase: viewcase,
    });
  };

  handleStatusChange = (modalstatus) => {
    this.setState({
      mstatus: modalstatus
    })
  }

  async componentDidMount() {
    console.log("in get all cases frontend")

    let agentID = localStorage.getItem("agentID");
    let organisationID = localStorage.getItem("organisationID")
    let count = 0
    axios.defaults.withCredentials = true;
    if(agentID){
      count = count+1
      let data = {
        agentID: agentID,
        organisationID: organisationID
      };
      console.log("data is..")
      console.log(data);
  
      axios({
        method: 'get',
        url: 'http://' + config.hostname + ':' + config.backendPort + '/getCases',
        params: data,
      })
        .then((response) => {
          console.log("response from getCases")
          console.log(response)
  
          if (response.status === 200) {
            let receivedCases = response.data.cases;
            console.log("received cases")
            console.log(response.data.cases)
            if (receivedCases) {
              this.setState({ allCases: receivedCases, flag: true });
            } else {
              this.setState({ allCases: {} });
            }
          }
          else {
            this.setState({ allCases: {} });
          }
  
        })
        .catch(error => {
          console.log(error);
          this.setState({ allCases: {} })
        })
    }
    else if(count < 1){
      window.location.reload();
    }
    
  }

  render() {
    const closeBtn = (
      <button className="close" onClick={() => this.showModal()}>
        &times;
       </button>
    );
    var casedetails;
    if (this.state.flag) {
      casedetails = this.state.allCases.map((ticket) => {
        return (
          <div>
            <div onClick={() => this.showModal1(ticket)} >
                  <Table>
                    <tbody >
                      <tr >
                        <td style={{ width: '10rem' }}>{ticket.CaseID}</td>
                        <td style={{ width: '15rem' }}>{ticket.Status}</td>
                        <td style={{ width: '50rem' }}>{ticket.Information} </td>
                      </tr>
                    </tbody>
                  </Table>            
            </div>
          </div>
        );
      });
    }
    return (
      <div>
        <NavbarDash />
        <div className="row">
          <div className="col-2">
            <Sidebar />
          </div>
          <div className="container col-9">  
          <form action="/action_page.php">
            <label for="filters">Filter by case status:   </label>
              <select name="status" id="status">
                <option value="New">New</option>
                <option value="Open">Open</option>
                <option value="Resolved">Resolved</option>
                <option value="Pending">Pending</option>
              </select>
            <br></br>
            <br></br>
            <br></br>
          </form>
          <Table>
          <tr>
            <th style={{ width: '10rem' }}>ID</th>
            <th style={{ width: '15rem' }}>Status</th>
            <th style={{ width: '50rem' }}>Details</th>
          </tr>
          </Table>
          {this.state.allCases.length > 0 ? (
              <div className="col-10">
                {casedetails}
                {this.state.viewcase != null ? (
                  <Modal
                    isOpen={this.state.modal}
                    toggle={() => this.showModal()}
                    className="modal-popup-lg"
                    scrollable
                  >
                    <ModalHeader
                      toggle={() => this.showModal()}
                      close={closeBtn}
                    >
                      Case Details
                   </ModalHeader>
                    <ModalBody className="modal-body">
                      <div className=" row form-group">
                        <p className="font-weight col-7">
                          Information : {this.state.viewcase.Information}
                        </p>
                        <div className="col-3">
                          <button class="btn btn-success dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                            {this.state.mstatus}</button>
                          <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
                            <button onClick={() => this.handleStatusChange("Working")} class="dropdown-item" >Working</button>
                            <button onClick={() => this.handleStatusChange("Resolved")} class="dropdown-item" >Resolved</button>
                            <button onClick={() => this.handleStatusChange("Postponed")} class="dropdown-item" >Postponed</button>
                          </div>
                        </div>
                      </div>
                      <div className="form-group">
                        <p className="font-weight-bold">
                          Resolution Comments : {this.state.viewcase.ResolutionComments}
                        </p>

                      </div>

                    </ModalBody>
                    <ModalFooter>
                    </ModalFooter>
                  </Modal>
                ) : null}
              </div>
            ) : (
                <div>
                  <h4 style={{ margin: "3em" }}>No new cases to display!</h4>
                </div>
              )}
          </div>

        </div>
      </div>
    );
  }
}

export default Dashboard;
