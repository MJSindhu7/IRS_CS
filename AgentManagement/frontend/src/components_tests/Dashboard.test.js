import React from 'react';
import renderer from 'react-test-renderer';
import AgentCaseDisplay from '../components/AgentCaseDisplay'
import Dashboard from '../components/Dashboard'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
configure({ adapter: new Adapter() });

import NavbarDash from "../components/NavbarDash";
import Sidebar from '../components/Sidebar'

describe('Calling Dashboard components correctly', () => {
    it('should render AgentCaseDisplay component correctly', () => {
      const component = shallow(<AgentCaseDisplay />);
      expect(component).toMatchSnapshot();
    });
    test('renders list-items, to display submit button', () => {
        const wrapper = shallow(<Dashboard />);
        expect(wrapper.find('SubmitButton')).toBeDefined();
      })
    it('should render NavbarDash component correctly', () => {
        const component = shallow(<NavbarDash />);
        expect(component).toMatchSnapshot();
      });
    it('should render Sidebar component correctly', () => {
        const component = shallow(<Sidebar />);
        expect(component).toMatchSnapshot();
      });
  });