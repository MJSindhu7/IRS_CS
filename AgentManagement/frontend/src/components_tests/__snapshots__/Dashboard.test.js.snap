// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calling AgentCaseDisplay component correctly should render AgentCaseDisplay component correctly 1`] = `ShallowWrapper {}`;

exports[`Calling AgentCaseDisplay component correctly should render NavbarDash component correctly 1`] = `ShallowWrapper {}`;

exports[`Calling AgentCaseDisplay component correctly should render Sidebar component correctly 1`] = `ShallowWrapper {}`;
